### PR TITLE
test: App Studio Plugin API combination and Unit Test Addition

### PR DESCRIPTION
**Pull request by yiqing-zhao**
Combine `getConfigForCreatingManifest()`, `getDevAppDefinition()`, `convertToAppDefinition()`, and `updateApp()`

Add unit test for app studio plugin. The coverage of app studio plugin is 70.38%.

### CODE REVIEW

File: packages/fx-core/tests/plugins/resource/appstudio/unit/createManifestForRemote.test.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    BotPlugin.name = "fx-resource-bot";
+    BotPlugin.displayName = "Bot";
+    selectedPlugins = [BotPlugin];
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it("maybeSelectedPlugins error", async () => {
```
Comment (1yefuwang1) [inline packages/fx-core/tests/plugins/resource/appstudio/unit/createManifestForRemote.test.ts:78]: BDD style tests are expected to be written in a way that a test can be read out. 
Something like it("should return error")

Comment (yiqing-zhao) [reply packages/fx-core/tests/plugins/resource/appstudio/unit/createManifestForRemote.test.ts]: Thanks for the reminder!  I have already change the testcase names.

**Review by 1yefuwang1 — APPROVED** [review]

File: packages/fx-core/src/plugins/resource/appstudio/index.ts [hunk 2]
```diff
@@ -77,23 +77,17 @@ export class AppStudioPlugin implements Plugin {
...
+  public async getAppDefinitionAndUpdate(
+    ctx: PluginContext,
     type: "localDebug" | "remote",
-    createIfNotExist: boolean,
-    teamsAppId?: string,
-    appStudioToken?: string,
-    logProvider?: LogProvider,
-    projectRoot?: string
+    manifest: TeamsAppManifest,
+    appStudioToken?: string
```
Comment (nliu-ms) [inline packages/fx-core/src/plugins/resource/appstudio/index.ts:84]: appStudioToken can also be removed from parameter, you can get this token from context.

Comment (yiqing-zhao) [reply packages/fx-core/src/plugins/resource/appstudio/index.ts]: Thanks! I have removed the appStudioToken from the parameter and get the token from the plugin Context.

**Review by nliu-ms — APPROVED** [review]

**Pull request merged**